### PR TITLE
Homework 2. Alexander Nalivayko. Cellular automaton.

### PR DESCRIPTION
**Pull request by AlexanderNalivayko**
Homework 2. Cellular automaton with just one life cycle.

### CODE REVIEW

**Comment by dir2000** [comment]
please enter number of columns: 5
Please enter number of rows: 7
Initialized matrix:
Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: 5
0 1 1 0 1 
0 1 0 1 1 
	at com.hillel.elementary.javageeks.cellular_automaton.MatrixProcessor.oneLifeCycle(Main.java:83)

**Comment by AlexanderNalivayko** [comment]
> please enter number of columns: 5
> Please enter number of rows: 7
> Initialized matrix:
> Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: 5
> 0 1 1 0 1
> 0 1 0 1 1
> at com.hillel.elementary.javageeks.cellular_automaton.MatrixProcessor.oneLifeCycle(Main.java:83)

Thank you :) I've fixed this issue in last commit.

**Comment by dir2000** [comment]
please enter number of columns: 0
Please enter number of rows: 1
Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: 0
	at com.hillel.elementary.javageeks.cellular_automaton.MatrixProcessor.fillMatrix(Main.java:47)
	at com.hillel.elementary.javageeks.cellular_automaton.Main.main(Main.java:31)

**Comment by AlexanderNalivayko** [comment]
> please enter number of columns: 0
> Please enter number of rows: 1
> Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: 0
> at com.hillel.elementary.javageeks.cellular_automaton.MatrixProcessor.fillMatrix(Main.java:47)
> at com.hillel.elementary.javageeks.cellular_automaton.Main.main(Main.java:31)

Thanks again for testing this code :) Resolved and fixed.

File: modules/alex-n/src/main/java/com/hillel/elementary/javageeks/cellular_automaton/Main.java [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+
+class MatrixProcessor{
+
+    //fill matrix with random 0 or 1
+    public static int[][] fillMatrix(int[][] matrix){
+
+        for (int i = 0; i < matrix[0].length; i++) {
+            for (int j = 0; j < matrix.length; j++) {
+                matrix[j][i] = (int) Math.round(Math.random());
```
Comment (dir2000) [inline modules/alex-n/src/main/java/com/hillel/elementary/javageeks/cellular_automaton/Main.java]: Method "nextInt()" of the "Random" class may be used too. For example: Random.nextInt(2)
https://www.tutorialspoint.com/java/util/random_nextint_inc_exc.htm

File: modules/alex-n/src/main/java/com/hillel/elementary/javageeks/cellular_automaton/Main.java [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        Scanner scanner = new Scanner(System.in);
+        boolean initializingData = true;
+
+        while (initializingData) {
+            System.out.printf("Please enter number of columns: ");
+            if (scanner.hasNextInt()) {
+                columns = scanner.nextInt();
+            } else {
+                System.out.printf("wrong data. Exit");
+                System.exit(0);
```
Comment (funtod) [inline modules/alex-n/src/main/java/com/hillel/elementary/javageeks/cellular_automaton/Main.java]: Exit code 0 means successful execution. Use 1

**Pull request merged**